### PR TITLE
Retrieve API Swagger Store REST API + UI

### PR DESCRIPTION
**Pull request by kavishkafernando**
## Purpose
Retrieve API Swagger Store REST API + UI
Fixing [issue/4661](https://github.com/wso2/product-apim/issues/4661)

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/components/Apis/Details/index.js [hunk 1]
```diff
@@ -95,12 +95,14 @@ class Details extends React.Component {
         let existing_subscriptions = api.getSubscriptions(this.api_uuid, null);
         let promised_applications = api.getAllApplications();
 
-        Promise.all([ promised_api, existing_subscriptions, promised_applications] ).then(
+        //Promise.all([ promised_api, existing_subscriptions, promised_applications] ).then(
```
Comment (npamudika) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/components/Apis/Details/index.js]: Shall we remove the commented lines?

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/components/Apis/Details/index.js [hunk 2]
```diff
@@ -95,12 +95,14 @@ class Details extends React.Component {
...
+        Promise.all([ promised_api] ).then(
             response => {
-                let [api, subscriptions, applications] = response.map(data => data.obj);
+                //let [api, subscriptions, applications] = response.map(data => data.obj);
```
Comment (npamudika) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/components/Apis/Details/index.js]: Same here

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/components/Apis/Details/index.js (hunk 1 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/components/Apis/Details/index.js]: The file has been updated with new logic and replaced with [index.jsx](https://github.com/wso2/carbon-apimgt/blame/6.5.x/features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/components/Apis/Details/index.jsx)

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/components/Apis/Details/index.js (hunk 2 above)
Comment (kavishkafernando) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/components/Apis/Details/index.js]: The file has been updated with new logic and replaced with [index.jsx](https://github.com/wso2/carbon-apimgt/blame/6.5.x/features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/components/Apis/Details/index.jsx)

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java [hunk 3]
```diff
@@ -228,10 +230,43 @@ public Response apisApiIdSdksLanguageGet(String apiId, String language) {
...
+    public Response apisApiIdSwaggerGet(String apiId, String ifNoneMatch, String xWSO2Tenant) {
+        String requestedTenantDomain = RestApiUtil.getRequestedTenantDomain(xWSO2Tenant);
+        try {
+            APIConsumer apiConsumer = RestApiUtil.getLoggedInUserConsumer();
+            if (!RestApiUtil.isTenantAvailable(requestedTenantDomain)) {
+                RestApiUtil.handleBadRequest("Provided tenant domain '" + xWSO2Tenant + "' is invalid", log);
+            }
+
+            //this will fail if user does not have access to the API or the API does not exist
+            APIIdentifier apiIdentifier = RestAPIStoreUtils.getAPIIdentifierFromUUID(apiId, requestedTenantDomain);;
```
Comment (AmaliMatharaarachchi) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java]: 2 ;s?

Comment (kavishkafernando) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java]: fixed

**Pull request merged**